File: app/src/main/java/wseemann/media/romote/fragment/StoreFragment.java
package wseemann.media.romote.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import wseemann.media.romote.R;

/**
 * Created by wseemann on 8/6/16.
 */
public class StoreFragment extends Fragment {

    private WebView mWebView;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        final View view = inflater.inflate(R.layout.fragment_store, container, false);
        mWebView = (WebView) view.findViewById(R.id.webview);

        return view;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        mWebView.loadUrl("https://channelstore.roku.com/browse");
        mWebView.setWebViewClient(new WebViewClient());
        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.getSettings().setJavaScriptCanOpenWindowsAutomatically(true);
    }

    public boolean onKeyDown(int keyCode, KeyEvent event) {
        // Check if the key event was the Back button and if there's history
        if ((keyCode == KeyEvent.KEYCODE_BACK) && mWebView.canGoBack()) {
            mWebView.goBack();
            return true;
        }
        // If it wasn't the Back key or there's no web page history, bubble up to the default
        // system behavior (probably exit the activity)
        return false;
    }
}

File: app/src/main/java/wseemann/media/romote/fragment/StoreFragment.kt
package wseemann.media.romote.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import dagger.hilt.android.AndroidEntryPoint
import wseemann.media.romote.composables.StoreScreen
import wseemann.media.romote.composables.theme.RomoteTheme
import wseemann.media.romote.viewmodels.StoreScreenViewModel

/**
 * Created by wseemann on 8/6/16.
 */
@AndroidEntryPoint
class StoreFragment : Fragment() {

    private lateinit var storeScreenViewModel: StoreScreenViewModel

    /**
     * Whether this fragment is the ViewPager's current page. MainActivity's SectionsPagerAdapter
     * uses the default FragmentPagerAdapter behavior (BEHAVIOR_SET_USER_VISIBLE_HINT) with an
     * offscreen page limit of 3, so this fragment is RESUMED even while another tab is on screen.
     * setUserVisibleHint is the signal that behavior mode drives, which is why the deprecated hook
     * is still the right one here - it replaces the old
     * `mViewPager.getCurrentItem() != 3` guard MainActivity used to apply before forwarding
     * back-key presses.
     */
    private var isCurrentPage by mutableStateOf(false)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        storeScreenViewModel = ViewModelProvider(this)[StoreScreenViewModel::class.java]
    }

    @Deprecated("Required by FragmentPagerAdapter's BEHAVIOR_SET_USER_VISIBLE_HINT mode")
    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        @Suppress("DEPRECATION")
        super.setUserVisibleHint(isVisibleToUser)
        isCurrentPage = isVisibleToUser
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext()).apply {
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
            setContent {
                val uiState by storeScreenViewModel.uiState.collectAsStateWithLifecycle()

                RomoteTheme {
                    StoreScreen(
                        uiState = uiState,
                        isCurrentPage = isCurrentPage,
                        onEvent = storeScreenViewModel::onHandleEvent
                    )
                }
            }
        }
    }
}
